### DISCUSSION TITLE
SDXL Takes 15 Minutes for a 1024px Image

### DISCUSSION THREAD

**Discussion by rekilla** [vladmandic/sdnext#2279] [Q&A]
I recently started using SD Next for image generation, but I've noticed that the process takes a substantial amount of time for generating a single image, despite having a system with 16GB RAM and a NVIDIA 3060 graphics card with 6GB VRAM. Previously, when I was using Comfy, the image generation took around 45 to 60 seconds. I'm wondering if there's a way to optimize the performance on SD Next to speed up the process. Is there a configuration or a setting that I might be missing, or is this the expected performance? Any guidance or suggestions would be greatly appreciated.

**Comment by vladmandic** [comment]
no, its not expected. my test system is rtx3060 and it generates 1024px with sdxl in 20-30sec tops.
but i cannot say anything more on your setup since i have no idea about it.

**Comment by rekilla** [comment]
i followed the instructions you gave and now it's fast like 40sec but im getting black outputs ,i can see the image while its generating ......how can i show you my setup didn't do anything or add anything

**Comment by vladmandic** [comment]
black image is one most commented on topics since first days of stable diffusion.
if you're using reference sdxl model, it has overflowing vae which doesn't work in fp16, so set vae to fp32. or get a fixed vae that works in fp16.

**Comment by rekilla** [comment]
thank you I did both fp32 and got vae fix ...works now
